sw/device/silicon_creator/lib/dice.c

```diff
@@ -319,3 +353,77 @@ rom_error_t dice_cdi_1_cert_build(manuf_certgen_inputs_t *inputs,
 
   return kErrorOk;
 }
+
+rom_error_t dice_tpm_ek_cert_build(manuf_certgen_inputs_t *inputs,
+                                   attestation_public_key_t *tpm_ek_pubkey,
+                                   uint8_t *tpm_ek_tbs, size_t *tpm_ek_tbs_size) {
+  hmac_digest_t digest;
```

Comment:
What is this used for? if you just need an array of 0 bytes maybe create one instead?

Reply:
`tpm_ek_build_tbs()` returns a digest of the pubkey of the certificate, it is not used in case of TPM certs, as there is no chain, but the function expects the digest space to fill, so I use the `hmac_digest_t` to make sure that the space is there. I could define it as `unsigned char buf[sizeiof(hmac_diget_t)];` instead, should I?

Reply:
Hmm, `tpm_ek_build_tbs()` does not return a digest of the pubkey, it needs the digest of the pubkey as an **_input_**, as the digest of the pubkey is what we use for the serial number for the DICE certs. If you are using the digest of the pubkey as the serial number for the TPM certs, then you need to pass in the digest of the pubkey to `dice_tpm_ek_cert_build()` since it is generated via the `tpm_attestation_keygen()` keygen function. Does that make sense?